### ISSUE TITLE
Any support for unions?

### ISSUE THREAD

**Issue by reureu** [charlesdevandiere/graphql-query-builder-dotnet#17]
Hi there,

Does GraphQL.Query.Builder support unions?
https://graphql.org/learn/schema/#union-types

If not, have you got any plan to support unions in a not too distant future?

Regards,
Reureu

**Comment by charlesdevandiere** [comment]
Hi @reureu,

GraphQL.Query.Builder doesn't support union, but it's a good idea.
I will implement it in the future, but I cannot give you a date.

**Comment by LoveWikberg** [comment]
Hi,

Great tool! Has there been any progress regarding the support for union types?

**Comment by charlesdevandiere** [comment]
Sorry for the wait. I just implemented union support in 2.2.0 #45

You can use:

```csharp
IQuery<TSource> AddUnion<TUnionType>(string typeName, Func<IQuery<TUnionType>, IQuery<TUnionType>> build)
```

or

```csharp
IQuery<TSource> AddUnion<TUnionType>(Func<IQuery<TUnionType>, IQuery<TUnionType>> build)
```
